internal/witness/cmd/witness/internal/witness/witness.go

```diff
@@ -104,14 +122,22 @@ func (w *Witness) GetLogs() ([]string, error) {
 	return logs, nil
 }
 
-// GetSTH gets an STH for a given log, which is consistent with all
+// GetSTH gets a cosigned STH for a given log, which is consistent with all
 // other STHs for the same log signed by this witness.
 func (w *Witness) GetSTH(logID string) ([]byte, error) {
-	sth, err := w.getLatestSTH(w.db.QueryRow, logID)
+	sthRaw, err := w.getLatestSTH(w.db.QueryRow, logID)
 	if err != nil {
 		return nil, err
 	}
-	return sth, nil
+	sth, err := w.parse(sthRaw, logID)
+	if err != nil {
+		return nil, fmt.Errorf("couldn't parse raw STH: %v", err)
+	}
+	signed, err := w.signSTH(sth)
```

Comment:
Is it worth a comment in here somewhere about why we changed to a "sign on the way out" approach?

Reply:
Sure!  I've added one right before `parse` since that's why I made the change, let me know what you think?